crates/types/src/retries.rs

```diff
@@ -124,25 +125,25 @@ impl Default for RetryPolicy {
 }
 
 impl RetryPolicy {
-    pub fn fixed_delay(interval: Duration, max_attempts: usize) -> Self {
+    pub fn fixed_delay(interval: Duration, max_attempts: Option<usize>) -> Self {
```

Comment:
Should `max_attempts` have the type `Option<NonZeroUsize>` to avoid the expect?

Reply:
All those were left in an attempt to stop the refactoring from cascading down tbh. My goal was to be able to ship this by end of day. I can definitely go over those types and follow the rabbit hole :)